src/main/java/org/opensearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -61,35 +80,28 @@ public boolean validate() {
                         this.errorType = ErrorType.INVALID_PASSWORD;
                         return false;
                     }
-
-                    if (!Strings.isNullOrEmpty(regex)) {
-                        // Password can be null for an existing user. Regex will validate password if present
-                        if (!Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                            if(log.isDebugEnabled()) {
-                                log.debug("Regex does not match password");
-                            }
-                            this.errorType = ErrorType.INVALID_PASSWORD;
-                            return false;
-                        }
-
-                        final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
-                        final boolean isDebugEnabled = log.isDebugEnabled();
-
+                    final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
+                    if (!Strings.isNullOrEmpty(regex) || useScoreBasedValidation) {
                         if (username == null || username.isEmpty()) {
-                            if (isDebugEnabled) {
+                            if (log.isDebugEnabled()) {
                                 log.debug("Unable to validate username because no user is given");
                             }
                             return false;
                         }
 
                         if (username.toLowerCase().equals(password.toLowerCase())) {
-                            if (isDebugEnabled) {
+                            if (log.isDebugEnabled()) {
                                 log.debug("Username must not match password");
                             }
                             this.errorType = ErrorType.INVALID_PASSWORD;
                             return false;
                         }
                     }
+                    if (!Strings.isNullOrEmpty(regex)) {
+                        return regexBasedPasswordValidation(regex, password);
+                    } else if (useScoreBasedValidation) {
```

Comment:
Could we allow for both validation paths?  Otherwise one could set `plugins.security.restapi.passwords.score.based.validation` and it would not be enforced.